### ISSUE TITLE
Can't set value for input[type=color] field

### ISSUE THREAD

**Issue by vfonic** [teamcapybara/capybara#2215]
## Meta
Capybara Version:
3.22.0
Driver Information (and browser if relevant):
Selenium::WebDriver::VERSION
3.142.3
Selenium::WebDriver::Remote::Capabilities.chrome.version
"" // I get empty string here in the CLI
From Chrome/About when navigating in the browser: 74.0.3729.169 (Official Build) (64-bit)
This is a rails 5.2.3 app, and I'm using rspec as a testing framework and rails system tests (not sure what exactly this means, but my tests are marked with `type: :system`).

```ruby
Capybara.server = :puma, { Silent: true }
Capybara.default_selector = :css
```

## Steps to reproduce

I've got an `input[type=color]` field.
I'm finding the element on the page like this:

```ruby
input_el = page.find("[name='#{input_name_field}']")
```

And then I'm setting its value (works for `input[type=text]` and `input[type=checkbox]`):

```ruby
input_el.set(value)
```

However, for `input[type=color]`, the Chrome color picker opens, instead of setting the value.

Here's the HTML (I'm not sure if it's relevant):

```html
<label for="arrows.color">
  Color<br>
  <input type="color" name="arrows.color" id="arrows.color" value="#000000"> 
  <input type="text" name="arrows.color" id="arrows.color" value="#000000"><br>
  <em></em>
</label>
```

(it contains both color picker and text input)

If I try to set color on the `input[type=text]`, that works fine:

```ruby
input_el = page.find("[name='#{input_name_field}'][type=text]")
input_el.set(value)
```

Any ideas what I could be doing wrong?

**Comment by twalpole** [comment]
Currently there is no special handling for color inputs, so whatever value you set is being sent as keystrokes. Unfortunately keystrokes aren't handled by color inputs and there is no way for Capybara/Selenium to interact with the dialog box that appears if we click on the color input.  For now, the only way to set the value of a color input would be via `execute_script`

    page.find("[name='#{input_name_field}']").execute_script('this.value = arguments[0]', value)

**Comment by vfonic** [comment]
Thanks! That workaround worked!

Do you think this should be mentioned somewhere in the docs?

**Comment by twalpole** [comment]
I'll probably implement it in the selenium driver today or tomorrow - so docs won't be necessary

**Comment by twalpole** [comment]
Closed via 8055941e0a531928159362e0a7b2c25ea1623b21
